example/operator/20-garden.yaml

```diff
@@ -196,6 +196,12 @@ spec:
     # gardenerScheduler:
     #   featureGates:
     #     SomeGardenerFeature: true
+    #   schedulerControllerConfiguration:
+    #     backupBucket:
+    #       concurrentSyncs: 5
+    #     shoot:
+    #       concurrentSyncs: 5
+    #       candidateDeterminationStrategy: SameRegion # either {SameRegion,MinimalDistance}
```

Comment:
As part of #8234, I was thinking about exposing this and decided against it - productive installations of Gardener (and the one deployed via `gardener-operator` is of such kind) would probably always set `MinimalDistance`. I would not expose this for now either to keep the API surface as small as possible, and wait and gather some feedback/experience until broadening it up (same approach we follow for `Shoot`s).

Reply:
I could imagine two scenarios where you want to use `SameRegion` instead of `MinimalDistance`:
- when you have rules like: "A Shoot must run in EU/US/... only"
- when you run a landscape with providers with a broken `MinimalDistance` calculation only. In this case you could use `SameRegion` for default and a custom scheduler for everything else.

Reply:
I guess the point is that probably nobody will run seed clusters in each and every region that is also offered for shoot clusters (too costly).

> when you have rules like: "A Shoot must run in EU/US/... only"

This can already be achieved with `seedSelector` and according labels on `Seed`s.

> when you run a landscape with providers with a broken `MinimalDistance` calculation only. In this case you could use `SameRegion` for default and a custom scheduler for everything else.

This is quite hypothetic and artificial, TBH. Custom scheduler support was only introduced a day ago. Nobody is using it. Still, in the end, with `SameRegion` people are enforced to provide seed clusters in all exposed shoot regions. Again, I don't think anybody will do it.

Look, I'm not fully against exposing this in the `Garden` API - I'm just thinking of not enlarging the API surface too much unnecessarily (since this always comes with additional validation, tests, wiring, hence complexity). Unless we see real feedback/necessity for it, I would simply spare these efforts. If you insist, we can still do it, but I doubt it provides much value for now.

Reply:
I don't have such a strong opinion for the to insist on an option for `candidateDeterminationStrategy`, so I removed it.